tutorial/unity_local_realm.md

```diff
@@ -0,0 +1,608 @@
+
+Let's first look into the initial setup of the board.
+
+## Setting up the board
+
+The first thing we want to do is to define a Realm representation of our Piece since we cannot save the `MonoBehaviour` directly in Realm. Classes that are supposed to be saved in Realm need to subclass `RealmObject`. The class `PieceEntity` will represent such an object. Note that we can not just duplicate the types from `Piece` since [not all of them can be saved in Realm](https://docs.mongodb.com/realm/sdk/dotnet/data-types/field-types/), like `Vector3` and `enum`.
+
+Add the following scripts to the project:
+
+```c#
+using Realms;
+using UnityEngine;
+
+public class PieceEntity : RealmObject
+{
+    // 1
+    public PieceType PieceType
+    {
+        get => (PieceType)Type;
+        private set => Type = (int)value;
+    }
+
+    // 2
+    public Vector3 Position
+    {
+        get => PositionEntity.ToVector3();
+        set => PositionEntity = new Vector3Entity(value);
+    }
+
+    // 3
+    private int Type { get; set; }
+    private Vector3Entity PositionEntity { get; set; }
+
+    // 4
+    public PieceEntity(PieceType type, Vector3 position)
+    {
+        PieceType = type;
+        Position = position;
+    }
+
+    // 5
+    protected override void OnPropertyChanged(string propertyName)
+    {
+        if (propertyName == nameof(PositionEntity))
+        {
+            RaisePropertyChanged(nameof(Position));
+        }
+    }
+
+    // 6
+    private PieceEntity()
+    {
+    }
+}
+```
+
+```c#
+using Realms;
+using UnityEngine;
+
+public class Vector3Entity : EmbeddedObject // 7
+{
+    public float X { get; private set; }
+    public float Y { get; private set; }
+    public float Z { get; private set; }
+
+    public Vector3Entity(Vector3 vector) // 8
+    {
+        X = vector.x;
+        Y = vector.y;
+        Z = vector.z;
+    }
+
+    public Vector3 ToVector3() => new Vector3(X, Y, Z); // 9
+
+    private Vector3Entity() // 10
+    {
+    }
+}
+```
+
+Even though we cannot save the `PieceType` (1) and the position (2) directly in the Realm, we can still expose them using backing variables (3) to make working with this class easier while still fulfilling the requirements for saving data in Realm.
+
+Additionally we provide a convenience constructor (4) for setting those two properties. A default constructor (6) also has to be provided for every `RealmObject`. Since we are not going to use it here though we can set it to `private`.
+
+Note that one of these backing variables is a `RealmObject` itself, or rather a sub class of it: `EmbeddedObject` (7). By extracting the position to a separate class `Vector3Entity` the `PieceEntity` is more readable. Another plus is that we can use the `EmbeddedObject` to represent a 1:1 relationship. Every `PieceEntity` can only have one `Vector3Entity` and even more importantly, ever `Vector3Entity` can only belong to one `PieceEntity` because there can only every be one `Piece` on any given `Square`.
```

Comment:
Note that one of these backing variables is a `RealmObject` itself, or rather a sub class of it: `EmbeddedObject` (7). By extracting the position to a separate class `Vector3Entity` the `PieceEntity` is more readable. Another plus is that we can use the `EmbeddedObject` to represent a 1:1 relationship. Every `PieceEntity` can only have one `Vector3Entity` and even more importantly, every `Vector3Entity` can only belong to one `PieceEntity` because there can only every be one `Piece` on any given `Square`.